youtube_dl/downloader/dash.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import unicode_literals
+from .common import FileDownloader
+from ..compat import compat_urllib_request
+
+import re
+
+
+class DashSegmentsFD(FileDownloader):
+    """
+    Download segments in a DASH manifest
+    """
+    def real_download(self, filename, info_dict):
+        self.report_destination(filename)
+        tmpfilename = self.temp_name(filename)
+        base_url = info_dict['url']
+        segment_urls = info_dict['segment_urls']
+
+        self.byte_counter = 0
+
+        def append_url_to_file(outf, target_url, target_name):
+            self.to_screen('[DashSegments] %s: Downloading %s' % (info_dict['id'], target_name))
+            req = compat_urllib_request.Request(target_url)
+            data = self.ydl.urlopen(req).read()
+            outf.write(data)
+            self.byte_counter += len(data)
+
+        def combine_url(base_url, target_url):
```

Comment:
A sample base_url and target_url is ```https://r2---sn-u5oxu-un5e.googlevideo.com/videoplayback/id/CsmdDsKjzN8.1/itag/135/source/yt_live_broadcast/requiressl/yes/playlist_type/CLEAN/gcr/tw/pmbypass/yes/ratebypass/yes/cmbypass/yes/mime/video%2Fmp4/live/1/gir/yes/upn/8EBB90O9gTY/fexp/9408142,9408288,9408420,9408710,9413503,9415304/signature/5F5F2D0E30C999428C51E2426A741ED87A7FC1B6.668FA2E6A1AA4986A3A77C9AACBB134929E608E9/sver/3/key/cms1/ip/140.112.247.145/ipbits/0/expire/1433448544/sparams/cmbypass,expire,gcr,gir,id,ip,ipbits,itag,live,mime,mm,mn,ms,mv,pl,playlist_type,pmbypass,ratebypass,requiressl,source/mm/32/mn/sn-u5oxu-un5e/ms/lv/mt/1433426887/mv/m/pl/20``` and ```sq/11/clen/683896/lmt/1431241064715340/dur/5.0```. The final URL should be ```https://r2---sn-u5oxu-un5e.googlevideo.com/videoplayback/id/CsmdDsKjzN8.1/itag/135/source/yt_live_broadcast/requiressl/yes/playlist_type/CLEAN/gcr/tw/pmbypass/yes/ratebypass/yes/cmbypass/yes/mime/video%2Fmp4/live/1/gir/yes/upn/8EBB90O9gTY/fexp/9408142,9408288,9408420,9408710,9413503,9415304/signature/5F5F2D0E30C999428C51E2426A741ED87A7FC1B6.668FA2E6A1AA4986A3A77C9AACBB134929E608E9/sver/3/key/cms1/ip/140.112.247.145/ipbits/0/expire/1433448544/sparams/cmbypass,expire,gcr,gir,id,ip,ipbits,itag,live,mime,mm,mn,ms,mv,pl,playlist_type,pmbypass,ratebypass,requiressl,source/mm/32/mn/sn-u5oxu-un5e/ms/lv/mt/1433426887/mv/m/pl/20/sq/11/clen/683896/lmt/1431241064715340/dur/5.0```, and urljoin gives a different result because base_url does not have a trailing slash.